test/assignment.coffee

```diff
@@ -185,6 +185,17 @@ test "#4787 destructuring of objects within arrays", ->
   eq b, arr[1].b
   deepEqual {a, b}, arr[1]
 
+test "#4798 destructuring of objects with splat within arrays", ->
+  arr = [1, {a:1, b:2}]
+  [...,{a, r...}] = arr
+  eq a, 1
+  deepEqual r, {b:2}
+  [b, {q...}] = arr
```

Comment:
This reuses `b` from the first test, it should get its own `test "#4798...` block.

Reply:
From the first assignment (`[...,{a, r...}] = arr`)  variables `a` and `r` are set.
The second assignment `[b, {q...}] = arr` sets variables `b` and `q`. 
I think `b` doesn't exist before the second assignment, so it's not reused.